src/packagedcode/citation.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import io
+import saneyaml
+from packagedcode import models
+
+"""
+Handle CITATION.cff files, see https://citation-file-format.github.io/
+https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md
+
+See https://github.com/citation-file-format/citation-file-format/tree/main/examples/1.2.0/pass for examples.
+"""
+
+class CitationHandler(models.DatafileHandler):
+    datasource_id = 'citation_cff'
+    default_package_type = 'citation'
+    path_patterns = ('*/CITATION.cff',)
+    description = 'Citation file format'
+    documentation_url = 'https://citation-file-format.github.io/'
+
+    @classmethod
+    def parse(cls, location):
+        metayaml = get_cff_data(location)
+        title = metayaml.get('title')
+        abstract = metayaml.get('abstract')
+        keywords = metayaml.get('keywords')
+        release_date = metayaml.get('date-released')
+        # This is the version of the software/item and not cff-version
+        version = metayaml.get('version')
+        parties = []
+        authors = metayaml.get('authors')
+        if authors:
+            for author in authors:
+                name = author.get('family-names') or author.get('name')
+                email = author.get('email')
+                url = author.get('orcid')
+                parties.append(models.Party(
+                    name=name,
+                    email=email,
+                    url=url
+                ))
+        repository_homepage_url = metayaml.get('repository')
+        repository_download_url = metayaml.get('repository-code')
+        extracted_license_statement = metayaml.get('license')

```

Comment:
There are interesting details in https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md#license